documentation/api/1.0/api.md

```diff
@@ -0,0 +1,864 @@
+	HypervisorMachineType string
+
+	// Debug changes the default hypervisor and kernel parameters to
+	// enable debug output where available.
+	Debug bool
+
+	// DefaultVCPUs specifies default number of vCPUs for the VM.
+	// Pod configuration VMConfig.VCPUs overwrites this.
+	DefaultVCPUs uint32
+
+	// DefaultMem specifies default memory size in MiB for the VM.
+	// Pod configuration VMConfig.Memory overwrites this.
+	DefaultMemSz uint32
+
+	// DefaultBridges specifies default number of bridges for the VM.
+	// Bridges can be used to hot plug devices
+	DefaultBridges uint32
+
+	// MemPrealloc specifies if the memory should be pre-allocated
+	MemPrealloc bool
+
+	// HugePages specifies if the memory should be pre-allocated from huge pages
+	HugePages bool
+
+	// Realtime Used to enable/disable realtime
+	Realtime bool
+
+	// Mlock is used to control memory locking when Realtime is enabled
+	// Realtime=true and Mlock=false, allows for swapping out of VM memory
+	// enabling higher density
+	Mlock bool
+
+	// DisableNestingChecks is used to override customizations performed
+	// when running on top of another VMM.
+	DisableNestingChecks bool
+}
+```
+
+##### `AgentType`
+```Go
+// AgentType describes the type of guest agent a Pod should run.
+type AgentType string
+
+const (
+	// NoopAgentType is the No-Op agent.
+	NoopAgentType AgentType = "noop"
+
+	// HyperstartAgent is the Hyper hyperstart agent.
+	HyperstartAgent AgentType = "hyperstart"
+
+	// KataContainersAgent is the Kata Containers agent.
+	KataContainersAgent AgentType = "kata"
+
+	// SocketTypeVSOCK is a VSOCK socket type for talking to an agent.
+	SocketTypeVSOCK = "vsock"
+
+	// SocketTypeUNIX is a UNIX socket type for talking to an agent.
+	// It typically means the agent is living behind a host proxy.
+	SocketTypeUNIX = "unix"
+)
+```
+
+##### `ProxyType`
+```Go
+// ProxyType describes a proxy type.
+type ProxyType string
+
+const (
+	// NoopProxyType is the noopProxy.
+	NoopProxyType ProxyType = "noopProxy"
+
+	// NoProxyType is the noProxy.
+	NoProxyType ProxyType = "noProxy"
+
+	// CCProxyType is the ccProxy.
+	CCProxyType ProxyType = "ccProxy"
+
+	// KataProxyType is the kataProxy.
+	KataProxyType ProxyType = "kataProxy"
+)
+```
+
+##### `ProxyConfig`
+```Go
+// ProxyConfig is a structure storing information needed from any
+// proxy in order to be properly initialized.
+type ProxyConfig struct {
+	Path  string
+	Debug bool
+}
+```
+
+##### `ShimType`
+```Go
+// ShimType describes a shim type.
+type ShimType string
+
+const (
+	// CCShimType is the ccShim.
+	CCShimType ShimType = "ccShim"
+
+	// NoopShimType is the noopShim.
+	NoopShimType ShimType = "noopShim"
+
+	// KataShimType is the Kata Containers shim type.
+	KataShimType ShimType = "kataShim"
+)
+```
+
+##### `NetworkModel`
+```Go
+// NetworkModel describes the type of network specification.
+type NetworkModel string
+
+const (
+	// NoopNetworkModel is the No-Op network.
+	NoopNetworkModel NetworkModel = "noop"
```

Comment:
It would be good to clarify if this means "no network" or "default network" or "fake network for testing purposes".